File: I1_MVPAndRetrofit/app/src/main/java/com/example/tsvetelinastoyanova/weatherreportrevisited/data/source/remote/GetDataService.java
package com.example.tsvetelinastoyanova.weatherreportrevisited.data.source.remote;

import com.example.tsvetelinastoyanova.weatherreportrevisited.model.WeatherObject;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GetDataService {

    @GET("weather")
    Call<WeatherObject> getWeatherForCity(@Query("APPID") String appId, @Query("q") String city);
}

File: I1_MVPAndRetrofit_Kotlin/app/src/main/java/com/example/tsvetelinastoyanova/weatherapp/data/source/remote/GetDataService.kt
package com.example.tsvetelinastoyanova.weatherapp.data.source.remote

import com.example.tsvetelinastoyanova.weatherapp.model.currentweather.CurrentWeatherObject
import com.example.tsvetelinastoyanova.weatherapp.model.forecast.ForecastObject
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface GetDataService {

    @GET("weather")
    fun getWeatherForCity(@Query("APPID") appId: String, @Query("q", encoded = false) city: String): Single<CurrentWeatherObject>

    @GET("forecast")
    fun getForecastForCity(@Query("APPID") appId: String, @Query("q", encoded = false) city: String): Single<ForecastObject>
}